_posts/2021-08-14-Prefer-Scoped-Enums-Over-Unscoped-Enums.md

```diff
@@ -0,0 +1,210 @@
+---
+layout: post
+title: "Prefer Scoped Enums Over Unscoped Enums (Notes)"
+date: 2021-08-14
+desc: "Notes on Item 10, Chapter 3 of Effective Modern C++"
+keywords: "Blog, C++, Enumerators, Enums, Notes"
+categories: [Blog, C++]
+tags: [Blog, C++]
+blog: [C++]
+published: true
+excerpt_separator: <!--more-->
+show_sidebar: false
+author: "Kushashwa Ravi Shrimali"
+icon: icon-html
+---
+
+## Scoped vs Unscoped Enums 
+
+* **General rule:** declaring a name inside curly braces is limited to that scope.
+* **Exception:** C++-98 style Enums
+
+<!--more-->
+
+---
+**NOTE**
+
+My notes on Chapter 3, Item 10 of Effective Modern C++ written by Scott Meyers.
+
+Some (or even all) of the text can be similar to what you see in the book, as these are notes: I've tried not to be unnecessarily creative with my words. :)
+
+---
+
+```cpp
+// You can't declare black, white, red in the scope containing the enum Color
+enum Color {
+    black, white, red;
+};
+
+auto white = false; // error: white already declared in this scope
+```
+
+* Unscoped Enums have implicit type conversions for their enumerators.
+* Enumerators can implicitly convert to integral types, and then to floating-point types.
+
+```cpp
+// Assume Color is declared like above
+Color c = red; // valid since Enumerator white is leaded to the scope Color is in
```

Comment:
What does `is leaded to the scope Color is in` mean?

Reply:
Ah, typo. Should be _leaked_, does that make sense?